tests/metrics/test_compare_ground_truth.py

```diff
@@ -0,0 +1,289 @@
+        "true_positives": 1,
+    },
+    "Zip": {
+        "false_negatives": 0,
+        "false_positives": 3,
+        "precision": 0.6666666666666666,
+        "recall": 1.0,
+        "true_positives": 6,
+    },
+}
+expected_overall_metrics = {
+    "true_positives": 17,
+    "false_positives": 14,
+    "false_negatives": 12,
+    "precision": (17 / (17 + 14)),
+    "recall": (17 / (17 + 12)),
+}
+
+
+@pytest.fixture(scope="function")
+def ex_cgt_object():
+    cgt_object = CompareGroundTruth(ground_truth_list, predictions_list)
+    cgt_object.set_all_label_metrics("overlap")
+    cgt_object.set_overall_metrics()
+    return cgt_object
+
+
+def test_labels(ex_cgt_object):
+    expected_labels.sort()
+    ex_cgt_object.labels.sort()
+    assert len(expected_labels) == len(ex_cgt_object.labels)
+    assert expected_labels == ex_cgt_object.labels
+
+
+def test_true_positives(ex_cgt_object):
+    for label in ex_cgt_object.labels:
+        print(label)
+        assert (
+            expected_all_label_metrics[label]["true_positives"]
+            == ex_cgt_object.all_label_metrics[label]["true_positives"]
+        )
+
+
+def test_false_positives(ex_cgt_object):
+    for label in ex_cgt_object.labels:
+        assert (
+            expected_all_label_metrics[label]["false_positives"]
+            == ex_cgt_object.all_label_metrics[label]["false_positives"]
+        )
+
+
+def test_false_negatives(ex_cgt_object):
+    for label in ex_cgt_object.labels:
+        assert (
+            expected_all_label_metrics[label]["false_negatives"]
+            == ex_cgt_object.all_label_metrics[label]["false_negatives"]
+        )
+
+
+@pytest.mark.parametrize(
+    "input_true_p, input_false_p, expected",
+    [
+        (10, 3, (10 / 13)),
+        (0, 0, 0),
+        (0, 5, 0),
+        (10, 0, 1),
+    ],
+)
+def test_precision(input_true_p, input_false_p, expected, ex_cgt_object):
+    assert expected == ex_cgt_object._get_precision(input_true_p, input_false_p)
+
+
+@pytest.mark.parametrize(
+    "input_true_p, input_false_n, expected",
+    [
+        (11, 2, (11 / 13)),
+        (0, 0, 0),
+        (0, 4, 0),
+        (11, 0, 1),
+    ],
+)
+def test_recall(ex_cgt_object, input_true_p, input_false_n, expected):
+    assert expected == ex_cgt_object._get_recall(input_true_p, input_false_n)
+
+
+def test_all_label_metrics(ex_cgt_object):
+    assert len(expected_all_label_metrics) == len(ex_cgt_object.all_label_metrics)
+    assert expected_all_label_metrics == ex_cgt_object.all_label_metrics
+
+
+def test_overall_label_metrics_true_p(ex_cgt_object):
+    assert (
+        ex_cgt_object.overall_metrics["true_positives"]
+        == expected_overall_metrics["true_positives"]
+    )
+
+
+def test_overall_label_metrics_false_p(ex_cgt_object):
+    assert (
+        ex_cgt_object.overall_metrics["false_positives"]
+        == expected_overall_metrics["false_positives"]
+    )
+
+
+def test_overall_label_metrics_false_n(ex_cgt_object):
+    assert (
```

Comment:
probably should change the way you have this setup anyways- but you could have just looped through the keys of expected_overall_metrics and made all of the assertions in one test vs. splitting it out into many tests